crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1418,11 +1386,7 @@ impl<'db> Type<'db> {
                 // are both of type Literal[345], for example.
                 false
             }
-            Type::SubclassOf(..) => {
-                // TODO once we have support for final classes, we can return `true` for some
-                // cases: type[C] is a singleton if C is final.
-                false
-            }
+            Type::SubclassOf(..) => false,
```

Comment:
I think we might thank ourselves for this in the future?
```suggestion
            // We eagerly transform `SubclassOf` to `ClassLiteral` for final types, so `SubclassOf` is never a singleton.
            Type::SubclassOf(..) => false,
```